### PR TITLE
feat: added support for claiming native fee

### LINKED ISSUE #47: [OmniBridge]: Add nativeFee into structures in the relayer

### PR DESCRIPTION
**Pull request by frolvanya**
closes #47,#48,#57,#58,#63

### CODE REVIEW

File: omni-relayer/src/workers/near.rs [hunk 1]
```diff
@@ -206,3 +332,77 @@ pub async fn claim_fee(redis_client: redis::Client, connector: Arc<OmniConnector
...
+    redis_client: redis::Client,
+    connector: Arc<OmniConnector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            utils::redis::NEAR_FIN_TRANSFER_EVENTS.to_string(),
```
Comment (olga24912) [inline omni-relayer/src/workers/near.rs:347]: Bad naming. 

When I read this code I feel like you ignore ClaimFeeEvent and handle the FinTransferEvent with nonce != 0 as well.

File: omni-relayer/src/workers/near.rs [hunk 2]
```diff
@@ -206,3 +332,77 @@ pub async fn claim_fee(redis_client: redis::Client, connector: Arc<OmniConnector
...
+        let mut handlers = Vec::new();
+        for (key, event) in events {
+            if let Ok(event) = serde_json::from_str::<Nep141LockerEvent>(&event) {
+                handlers.push(tokio::spawn({
+                    let config = config.clone();
+                    let mut redis_connection = redis_connection.clone();
+                    let connector = connector.clone();
+
+                    async move {
+                        let Nep141LockerEvent::FinTransferEvent {
```
Comment (olga24912) [inline omni-relayer/src/workers/near.rs:367]: How it going to work in case of ClaimFeeEvent?

File: omni-relayer/src/workers/evm.rs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    redis_client: redis::Client,
+    connector: Arc<OmniConnector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            utils::redis::ETH_WITHDRAW_EVENTS.to_string(),
```
Comment (olga24912) [inline omni-relayer/src/workers/evm.rs:175]: Why ETH_WITHDRAW_EVENTS not SignClaimNativeFeeEvent?

**Review by olga24912 — CHANGES_REQUESTED** [review]
This PR is huge and contains a lot of independent changes. Could you split it into a few smaller ones? It would be much easier to review.

File: omni-relayer/src/config.rs [hunk 4]
```diff
@@ -26,13 +27,15 @@ pub struct Near {
...
 
 #[derive(Debug, Clone, serde::Deserialize)]
-pub struct Eth {
+pub struct Evm {
     pub rpc_http_url: String,
     pub rpc_ws_url: String,
     pub chain_id: u64,
     pub bridge_token_factory_address: Address,
     pub block_processing_batch_size: u64,
+    pub relayer: OmniAddress,
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/src/config.rs:40]: TODO: this address should be derived from private key

File: omni-relayer/Cargo.toml [hunk 5]
```diff
@@ -15,22 +15,25 @@ dotenv = "0.15"
...
 near-jsonrpc-client = "0.13.0"
 near-jsonrpc-primitives = "0.26.0"
 near-primitives = "0.26.0"
 near-crypto = "0.26.0"
-eth-proof = { git = "https://github.com/Near-One/rainbow-bridge-sdk-rs", branch = "omni-token-support", package = "eth-proof" }
-omni-connector = { git = "https://github.com/Near-One/rainbow-bridge-sdk-rs", branch = "omni-token-support", package = "omni-connector" }
+# eth-proof = { git = "https://github.com/Near-One/rainbow-bridge-sdk-rs", branch = "omni-token-support", package = "eth-proof" }
+eth-proof = { path = "../../rainbow-bridge-sdk-rs/bridge-sdk/eth-proof" }
+# omni-connector = { git = "https://github.com/Near-One/rainbow-bridge-sdk-rs", package = "omni-connector" }
+omni-connector = { path = "../../rainbow-bridge-sdk-rs/bridge-sdk/connectors/omni-connector" }
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/Cargo.toml:34]: Fixed in 4313488

File: omni-relayer/src/workers/evm.rs (hunk 3 above)
Comment (frolvanya) [reply omni-relayer/src/workers/evm.rs:175]: Fixed in 0f3efd2b

File: omni-relayer/src/workers/near.rs (hunk 2 above)
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs:367]: Fixed in 88245314c0985fcd6f219dcb57ddce4b3f1ad72c

**Pull request closed without merging**